File: whistleblower/workflows-java/src/test/java/net/corda/examples/whistleblower/FlowTests.java
package net.corda.examples.whistleblower;

import com.google.common.collect.ImmutableList;
import net.corda.core.concurrent.CordaFuture;
import net.corda.core.transactions.SignedTransaction;
import net.corda.examples.whistleblower.flows.BlowWhistleFlow;
import net.corda.examples.whistleblower.flows.BlowWhistleFlowResponder;
import net.corda.testing.node.MockNetwork;
import net.corda.testing.node.MockNetworkParameters;
import net.corda.testing.node.StartedMockNode;
import net.corda.testing.node.TestCordapp;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

public class FlowTests {
    private final MockNetwork network = new MockNetwork(new MockNetworkParameters()
            .withCordappsForAllNodes(ImmutableList.of(
                    TestCordapp.findCordapp("net.corda.examples.whistleblower.contracts"),
                    TestCordapp.findCordapp("net.corda.examples.whistleblower.flows"))));
    private final StartedMockNode a = network.createNode();
    private final StartedMockNode b = network.createNode();
    private final StartedMockNode c = network.createNode();

    public FlowTests() {
        ImmutableList.of(a, b).forEach(it -> it.registerInitiatedFlow(BlowWhistleFlowResponder.class));
    }

    @Before
    public void setUp() throws Exception {
        network.runNetwork();
    }

    @After
    public void tearDown() throws Exception {
        network.stopNodes();
    }

    //simple unit test to check the public keys that used in the transaction
    //are different from both mock node a's legal public key and mock node b's legal public key.
    @Test
    public void dummyTest() throws ExecutionException, InterruptedException {
        CordaFuture<SignedTransaction> future = a.startFlow(new BlowWhistleFlow(b.getInfo().getLegalIdentities().get(0), c.getInfo().getLegalIdentities().get(0)));
        network.runNetwork();
        SignedTransaction ptx = future.get();
        assert (!ptx.getTx().getRequiredSigningKeys().contains(a.getInfo().getLegalIdentities().get(0).getOwningKey()));
        assert (!ptx.getTx().getRequiredSigningKeys().contains(b.getInfo().getLegalIdentities().get(0).getOwningKey()));
    }
}

File: whistleblower/workflows-kotlin/src/test/kotlin/net/corda/examples/whistleblower/FlowTests.kt
package net.corda.examples.whistleblower

import net.corda.examples.whistleblower.flows.BlowWhistleFlow
import net.corda.examples.whistleblower.flows.BlowWhistleFlowResponder
import net.corda.testing.node.MockNetwork
import net.corda.testing.node.MockNetworkParameters
import net.corda.testing.node.TestCordapp
import org.junit.After
import org.junit.Before
import org.junit.Test

class FlowTests {
    private val network = MockNetwork(MockNetworkParameters(cordappsForAllNodes = listOf(
        TestCordapp.findCordapp("net.corda.examples.whistleblower.contracts"),
        TestCordapp.findCordapp("net.corda.examples.whistleblower.flows")
    )))
    private val a = network.createNode()
    private val b = network.createNode()
    private val c = network.createNode()

    init {
        listOf(a, b).forEach {
            it.registerInitiatedFlow(BlowWhistleFlowResponder::class.java)
        }
    }

    @Before
    fun setup() = network.runNetwork()

    @After
    fun tearDown() = network.stopNodes()

    //simple unit test to check the public keys that used in the transaction
    //are different from both mock node a's legal public key and mock node b's legal public key.
    @Test
    fun `dummy test`() {
        val future = a.startFlow(BlowWhistleFlow(b.info.legalIdentities.first(),c.info.legalIdentities.first()))
        network.runNetwork()
        val ptx = future.get()
        assert(!ptx.tx.requiredSigningKeys.contains(a.info.legalIdentities.first().owningKey))
        assert(!ptx.tx.requiredSigningKeys.contains(b.info.legalIdentities.first().owningKey))

    }
}